pkg/devfile/adapters/common/command.go

```diff
@@ -170,6 +184,11 @@ func GetInitCommand(data data.DevfileData, devfileInitCmd string) (initCommand c
 	return getCommand(data, devfileInitCmd, common.InitCommandGroupType)
 }
 
+// GetCommandByName iterates through the components in the devfile and returns the command with the given name
+func GetCommandByName(data data.DevfileData, desiredCommand string) (command common.DevfileCommand, err error) {
```

Comment:
This was really handy, thank you John :) A much nicer implementation. I've had to take it out of adapter at least for now, perhaps it will move if/when we reach a solution on the execDevfileEvent conversation.